addon-test-support/helpers.js

```diff
@@ -110,30 +117,100 @@ function buildKeyboardEvent(type, options = {}) {
   return event;
 }
 
+/*
+  @method click
+  @param {String} selector
+  @param {Object} options
+  @return {RSVP.Promise}
+  @public
+*/
 export function click(selector, options = {}) {
-  let el = document.querySelector(selector);
+  let el = first(selector);
   run(() => fireEvent(el, 'mousedown', options));
   focus(el);
   run(() => fireEvent(el, 'mouseup', options));
   run(() => fireEvent(el, 'click', options));
   return wait();
 }
 
+/*
+  @method fillIn
+  @param {String} selector
+  @param {String} text
+  @return {RSVP.Promise}
+  @public
+*/
 export function fillIn(selector, text) {
-  let el = document.querySelector(selector);
+  let el = first(selector);
   run(() => focus(el));
   run(() => el.value = text);
   run(() => fireEvent(el, 'input'));
   run(() => fireEvent(el, 'change'));
   return wait();
 }
 
+/*
+  @method triggerEvent
+  @param {String} selector
+  @param {String} type
+  @param {Object} options
+  @return {RSVP.Promise}
+  @public
+*/
 export function triggerEvent(selector, type, options) {
-  let el = document.querySelector(selector);
+  let el = first(selector);
   run(() => fireEvent(el, type, options));
   return wait();
 }
 
+/*
+  @method keyEvent
+  @param {String} selector
+  @param {String} type
+  @param {Number} keyCode
+*/
 export function keyEvent(selector, type, keyCode) {
   return triggerEvent(selector, type, { keyCode, which: keyCode });
 }
+
+/*
+  The find test helper uses `querySelectorAll` to search inside the test
+  DOM (based on app configuration for the rootElement).
+
+  Alternalively, a second argument may be passed which is an element as the
+  DOM context to search within.
+
+  @method find
+  @param {String} CSS selector to find one or more elements in the test DOM
+  @param {HTMLElement} contextEl to query within, query from its contained DOM
+  @return {HTMLElement|NodeList} one element or a (non-live) list of element objects
+  @public
+*/
+export function find(selector, contextEl) {
+  if (selector instanceof HTMLElement || selector instanceof NodeList) {
+    return selector;
+  }
+  let list;
+  if (contextEl instanceof HTMLElement) {
+    list = contextEl.querySelectorAll(selector);
+  } else {
+    list = document.querySelectorAll(`${rootElement} ${selector}`);
+  }
+  return (list.length === 1) ? list[0] : list;
+}
+
+
+/*
+  Since the find helper may return a NodeList or an HTMLElement the first helper
+  only returns the first matching element
+
+  @method first
+  @param {String} CSS selector to find one element in the test DOM
+  @param {HTMLElement} contextEl to query within, query from its contained DOM
+  @return {HTMLElement} first element found with matching selector
+  @public
+*/
+export function first(selector, contextEl) {
+  const el = find(selector, contextEl);
+  return (el instanceof NodeList) ? el[0] : el;
```

Comment:
Hmm, seems like this should be handled in `find`, no?  Meaning, instead of just returning a `NodeList` if passed in, we would return `selector[0]`...